### PR TITLE
Reviews interop tests naming

### PR DESCRIPTION
**Pull request by ignasi35**
Standardize the spec names.

Introduce `Servers` and `Clients`.

### CODE REVIEW

File: interop-tests/src/test/scala/akka/grpc/interop/GrpcInteropSpec.scala [hunk 1]
```diff
@@ -6,25 +6,49 @@ package akka.grpc.interop
...
+class GrpcInteropAkkaScalaWithAkkaNettyJavaSpec extends GrpcInteropTests(Servers.Akka.Scala, Clients.AkkaNetty.Java)
+// TODO testing against grpc-java server (problem with the path, still to be diagnosed)
+class GrpcInteropAkkaScalaWithAkkaHttpScalaSpec extends GrpcInteropTests(Servers.Akka.Scala, Clients.AkkaHttp.Scala)
+//class GrpcInteropAkkaScalaWithAkkaHttpJavaSpec extends GrpcInteropTests(Servers.Akka.Scala, Clients.AkkaHttp.Java)
+
+class GrpcInteropAkkaJavaWithIoSpec extends GrpcInteropTests(Servers.Akka.Java, Clients.IoGrpc)
+class GrpcInteropAkkaJavaWithAkkaNettyScalaSpec extends GrpcInteropTests(Servers.Akka.Java, Clients.AkkaNetty.Scala)
+class GrpcInteropAkkaJavaWithAkkaNettyJavaSpec extends GrpcInteropTests(Servers.Akka.Java, Clients.AkkaNetty.Java)
+//class GrpcInteropAkkaJavaWithAkkaHttpScalaSpec extends GrpcInteropTests(Servers.Akka.Java, Clients.AkkaHttp.Scala)
+//class GrpcInteropAkkaJavaWithAkkaHttpJavaSpec extends GrpcInteropTests(Servers.Akka.Java, Clients.AkkaHttp.Java)
```
Comment (ignasi35) [inline interop-tests/src/test/scala/akka/grpc/interop/GrpcInteropSpec.scala:26]: Using the aliases from `Clients` and `Servers` makes it a bit more readable.

I've also added a commented line for the missing specs that will use `Client.AkkaHttp.Java`.

File: interop-tests/src/test/scala/akka/grpc/interop/GrpcInteropSpec.scala [hunk 2]
```diff
@@ -6,25 +6,49 @@ package akka.grpc.interop
...
+class GrpcInteropAkkaScalaWithAkkaNettyScalaSpec extends GrpcInteropTests(Servers.Akka.Scala, Clients.AkkaNetty.Scala)
+class GrpcInteropAkkaScalaWithAkkaNettyJavaSpec extends GrpcInteropTests(Servers.Akka.Scala, Clients.AkkaNetty.Java)
+// TODO testing against grpc-java server (problem with the path, still to be diagnosed)
+class GrpcInteropAkkaScalaWithAkkaHttpScalaSpec extends GrpcInteropTests(Servers.Akka.Scala, Clients.AkkaHttp.Scala)
+//class GrpcInteropAkkaScalaWithAkkaHttpJavaSpec extends GrpcInteropTests(Servers.Akka.Scala, Clients.AkkaHttp.Java)
+
+class GrpcInteropAkkaJavaWithIoSpec extends GrpcInteropTests(Servers.Akka.Java, Clients.IoGrpc)
+class GrpcInteropAkkaJavaWithAkkaNettyScalaSpec extends GrpcInteropTests(Servers.Akka.Java, Clients.AkkaNetty.Scala)
+class GrpcInteropAkkaJavaWithAkkaNettyJavaSpec extends GrpcInteropTests(Servers.Akka.Java, Clients.AkkaNetty.Java)
+//class GrpcInteropAkkaJavaWithAkkaHttpScalaSpec extends GrpcInteropTests(Servers.Akka.Java, Clients.AkkaHttp.Scala)
```
Comment (ignasi35) [inline interop-tests/src/test/scala/akka/grpc/interop/GrpcInteropSpec.scala:25]: `GrpcInteropAkkaJavaWithAkkaHttpScalaSpec` is commented becuase there's a failure. I'll look into that in a separate PR.

Comment (raboof) [reply interop-tests/src/test/scala/akka/grpc/interop/GrpcInteropSpec.scala:25]: Good find, surprising!

File: interop-tests/src/test/scala/akka/grpc/interop/GrpcInteropSpec.scala [hunk 3]
```diff
@@ -6,25 +6,49 @@ package akka.grpc.interop
...
+class GrpcInteropIoWithIoSpec extends GrpcInteropTests(Servers.IoGrpc, Clients.IoGrpc)
+class GrpcInteropIoWithAkkaNettyScalaSpec extends GrpcInteropTests(Servers.IoGrpc, Clients.AkkaNetty.Scala)
+class GrpcInteropIoWithAkkaNettyJavaSpec extends GrpcInteropTests(Servers.IoGrpc, Clients.AkkaNetty.Java)
+class GrpcInteropIoWithAkkaHttpScalaSpec extends GrpcInteropTests(Servers.IoGrpc, Clients.AkkaHttp.Scala)
+//class GrpcInteropIoWithAkkaHttpJavaSpec extends GrpcInteropTests(Servers.IoGrpc, Clients.AkkaHttp.Java)
+
+class GrpcInteropAkkaScalaWithIoSpec extends GrpcInteropTests(Servers.Akka.Scala, Clients.IoGrpc)
+class GrpcInteropAkkaScalaWithAkkaNettyScalaSpec extends GrpcInteropTests(Servers.Akka.Scala, Clients.AkkaNetty.Scala)
+class GrpcInteropAkkaScalaWithAkkaNettyJavaSpec extends GrpcInteropTests(Servers.Akka.Scala, Clients.AkkaNetty.Java)
+// TODO testing against grpc-java server (problem with the path, still to be diagnosed)
```
Comment (raboof) [inline interop-tests/src/test/scala/akka/grpc/interop/GrpcInteropSpec.scala:18]:
```suggestion
```

**Review by raboof — APPROVED** [review]
one comment that doesn't look like it applies anymore.

**Comment by raboof** [comment]
(failure was #935, restarted)

File: interop-tests/src/test/scala/akka/grpc/interop/GrpcInteropSpec.scala (hunk 2 above)
Comment (ignasi35) [reply interop-tests/src/test/scala/akka/grpc/interop/GrpcInteropSpec.scala:24]: Turns out AkkaJava and AkkaScala respond to an `unimplemented_service` differently. See #1194

**Pull request merged**